documentation/adr/004-multi-LZA-configuration-management.md

```diff
@@ -0,0 +1,44 @@
+---
+layout: page.njk
+status: Proposed
+date: git Created
+tags:
+  - adr
+order: 4
+title: 004 - Enable Multi LZA Configuration 
+---
+
+**Date: {{ page.date.toISOString().substring(0,10) }}**
+
+**Status: {{ status }}**
+
+## Context
+
+To independently develop and assure features and functionality before general availability, multiple LZA environments are required as part of the teams SDLC. As a result, the team need a way to manage environment specific configuration e.g., workload accounts.
+
+The files used within LZA to manage the configuration of the deployment are YAML format, with the exception being IAM policies that are JSON. These are not required to differ between deployments.
```

Comment:
We may need to vary these i.e. test them in our dev account. However I expect we can manage this but creating new policies and the overriding the reference to the file in the yaml. We should probably acknowledge that possibly as we have before as a consequence e.g

- we can only template yaml files
- to replace files in other formats a complete file must be provided and the reference to the file be overridden in the yaml.
- **environment specific configuration will be deployed to codecommit repositories in environments they're not targeted for**

I put the last one in bold because it should be included anyway

Reply:
Updated in latest commit.